text/0000-lts-release-strategy.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: default
+title: Fabric LTS release strategy
+nav_order: 3
+---
+
+- Feature Name: Fabric LTS release strategy
+- Start Date: 2020-03-18
+- RFC PR:
+- Fabric Component:
+- Fabric Issue:
+
+# Summary
+[summary]: #summary
+
+Define Long-term support (LTS) release strategy for Hyperledger Fabric
+
+Long-term support definition from wikipedia.org:
+
+> **Long-term support (LTS)** is a product lifecycle management policy in which a stable release of computer software is maintained for a longer period of time than the standard edition.
+
+> **LTS** applies the tenets of reliability engineering to the software development process and software release life cycle. Long-term support extends the period of software maintenance; it also alters the type and frequency of software updates (patches) to reduce the risk, expense, and disruption of software deployment, while promoting the dependability of the software.
+
+# Motivation
+[motivation]: #motivation
+
+Administrators of production deployments find LTS releases desirable as they provide confidence that a certain software release will be maintained for an extended period of time.
+
+Specifically, for Hyperledger Fabric, administrators can expect the following of a LTS release:
+
+- Stable and well-tested code
+- API contract stability
```

Comment:
Are we suggesting that there is a possibility for the contract API to change with each LTS release but not within an LTS? Many of the platforms I have used are very strict about the APIs exposed to the user. Corda has promised that it would never break backward compatibility. Many databases provide that guarantee. Only programming languages have not promised such a compatibility guarantee.  

Even minor changes like moving chaincode to `fabric-chaincode-go`, `fabric-proto-go` repos as well as changing the `logger` have frustrated a few internal users as they have to change the chaincode (good that it was just experimental, not production). I think we made this hard decision considering the code maintenance in the long term but I doubt whether this would happen frequently.

Hence, I am not sure whether we are conveying the wrong message in this bullet point.

Reply:
A random philosophical thought of the day (couldn't resist me from posting this): 
I could call Fabric as a decentralized and replicated blockchain database.

1. Every peer is a replica of one other and smart-contract is more like a SQL procedure sitting close to the data. 
2. It is decentralized because there is no central authority. 
3. It is blockchain because it provides non-repudiation, immutability, provenance, etc..

If we hide endorsement knowledge from the client that submits the transaction and lets the peer or SDK library collect it, client would feel as if he/she is using a database such as PostgreSQL (i.e., invoking a stored SQL procedure) or MongoDB. 

APIs exposed are `GetXXX()` and `PutXXX()` for writing smart-contract. It is more like a database APIs. If these APIs are bad, we can change it in the next LTS. Going forward, it would be very nice to keep a stable API to keep our users happy. Users who have experience in using databases would appreciate it.

Reply:
There is no suggestion that APIs will change across LTS releases. That is not something we want to do. However given the amount of innovation in the blockchain space, we can't predict what changes future maintainers will want to make.

Reply:
> I think there are a few lines which could be interpreted slightly differently

I am one of them who interpreted differently (like Gari mentioned). 

What does this API Contract Stability refer too? Code stability? or existing API changes? I understood it as existing APIs wouldn't change within an LTS.

I agree that appending APIs for a new feature is no issue. Every databases/products do that.